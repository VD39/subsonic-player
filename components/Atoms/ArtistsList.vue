<script setup lang="ts">
defineProps<{
  artists: BaseArtist[];
}>();
</script>

<template>
  <ul class="commaList">
    <li
      v-for="artist in artists"
      :key="artist.id"
      data-test-id="artist-list-item"
    >
      <NuxtLink :to="`/artist/${artist.id}`" class="link">
        {{ artist.name }}
      </NuxtLink>
    </li>
  </ul>
</template>
