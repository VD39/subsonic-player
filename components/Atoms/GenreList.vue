<script setup lang="ts">
defineProps<{
  genres: Genre[];
}>();
</script>

<template>
  <ul class="commaList">
    <li
      v-for="genre in genres"
      :key="genre.name"
      data-test-id="genre-list-item"
    >
      <NuxtLink
        :aria-label="`View all albums with ${genre.name} genre`"
        class="link"
        :to="`/genre/albums/${encodeURIComponent(genre.name)}`"
      >
        {{ genre.name }}
      </NuxtLink>
    </li>
  </ul>
</template>
