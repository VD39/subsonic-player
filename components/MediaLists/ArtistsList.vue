<script setup lang="ts">
import NoMediaMessage from '@/components/NoMediaMessage/NoMediaMessage.vue';
import MediaListWrapper from './MediaListWrapper.vue';
import ArtistItem from './ArtistItem.vue';

defineProps<{
  artists: Artist[];
  rows?: string;
}>();
</script>

<template>
  <MediaListWrapper v-if="artists.length" :row="rows">
    <ArtistItem v-for="artist in artists" :key="artist.id" :artist="artist" />
  </MediaListWrapper>
  <NoMediaMessage v-else icon="PhUsersThree" message="No artists found." />
</template>
