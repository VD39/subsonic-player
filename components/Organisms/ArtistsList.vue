<script setup lang="ts">
import MediaListWrapper from '@/components/Atoms/MediaListWrapper.vue';
import NoMediaMessage from '@/components/Atoms/NoMediaMessage.vue';
import ArtistItem from '@/components/Organisms/ArtistItem.vue';

defineProps<{
  artists: Artist[] | SimilarArtist[];
}>();
</script>

<template>
  <MediaListWrapper v-if="artists.length">
    <ArtistItem v-for="artist in artists" :key="artist.id" :artist />
  </MediaListWrapper>

  <NoMediaMessage
    v-else
    :icon="IMAGE_DEFAULT_BY_TYPE.artist"
    message="No artists found."
  />
</template>
