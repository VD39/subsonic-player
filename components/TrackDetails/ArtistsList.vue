<script setup lang="ts">
defineProps<{
  artists: BaseArtist[];
}>();
</script>

<template>
  <ul :class="[$style.meta, $style.list]">
    <li
      v-for="artist in artists"
      :key="artist.id"
      :class="$style.listItem"
      data-test-id="artist-list-item"
    >
      <NuxtLink :to="`/artist/${artist.id}`">
        {{ artist.name }}
      </NuxtLink>
    </li>
  </ul>
</template>

<style module>
.list {
  @mixin align-center;
}

.listItem {
  &:not(:last-child)::after {
    content: ',\00a0';
  }

  &:nth-last-child(2)::after {
    content: '\00a0&\00a0';
  }
}
</style>
