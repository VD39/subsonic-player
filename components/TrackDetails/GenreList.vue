<script setup lang="ts">
defineProps<{
  genres: Genre[];
}>();
</script>

<template>
  <ul :class="[$style.meta, $style.list]">
    <li
      v-for="genre in genres"
      :key="genre.name"
      :class="$style.listItem"
      data-test-id="genre-list-item"
    >
      <NuxtLink :to="`/genre/albums/${genre.name}`">
        {{ genre.name }}
      </NuxtLink>
    </li>
  </ul>
</template>

<style module>
.list {
  @mixin align-center;
}

.listItem {
  &:not(:last-child)::after {
    content: ',\00a0';
  }
}
</style>
