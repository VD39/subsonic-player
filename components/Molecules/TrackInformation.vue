<script setup lang="ts">
import ArtistsList from '@/components/Atoms/ArtistsList.vue';
import GenreList from '@/components/Atoms/GenreList.vue';
import LinkOrText from '@/components/Atoms/LinkOrText.vue';

defineProps<{
  track: Track;
}>();
</script>

<template>
  <div :class="['trackTable', $style.trackInformation]">
    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Title</div>

      <div class="trackCell">
        <p>{{ track.name }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Album</div>

      <div class="trackCell">
        <LinkOrText
          :is-link="!!track.albumId"
          :text="track.album"
          :to="`/album/${track.albumId}`"
        />
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Artists</div>

      <div class="trackCell">
        <ArtistsList
          v-if="track.artists.length"
          :artists="track.artists"
          :class="$style.list"
        />

        <p v-else ref="artistsElse">{{ DEFAULT_VALUE }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Track Number</div>

      <div class="trackCell">
        <p>{{ track.trackNumber }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Duration</div>

      <time class="trackCell">
        {{ track.duration }}
      </time>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Year</div>

      <div class="trackCell">
        <p>{{ track.year }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Disc number</div>

      <div class="trackCell">
        <p>{{ track.discNumber }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Genre</div>

      <div class="trackCell">
        <GenreList
          v-if="track.genres.length"
          :class="$style.list"
          :genres="track.genres"
        />

        <p v-else ref="genresElse">{{ DEFAULT_VALUE }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Play count</div>

      <div class="trackCell">
        <p>{{ track.information.playCount }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Size</div>

      <div class="trackCell">
        <p>{{ track.size }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Content type</div>

      <div class="trackCell">
        <p>
          {{ track.information.contentType }}
        </p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Suffix</div>

      <div class="trackCell">
        <p>
          {{ track.information.suffix }}
        </p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Bitrate</div>

      <div class="trackCell">
        <p>{{ track.information.bitRate }}</p>
      </div>
    </div>

    <div class="trackRow trackBorder">
      <div :class="['trackCell', 'trackMeta', $style.title]">Path</div>

      <div class="trackCell">
        <p>{{ track.information.path }}</p>
      </div>
    </div>
  </div>
</template>

<style module>
.trackInformation {
  a {
    color: var(--theme-color);
  }
}

.title {
  width: 25%;
}

.list {
  gap: 0;
}
</style>
