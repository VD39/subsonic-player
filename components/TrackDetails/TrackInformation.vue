<script setup lang="ts">
import ArtistsList from '@/components/TrackDetails/ArtistsList.vue';
import GenreList from '@/components/TrackDetails/GenreList.vue';

defineProps<{
  track: Track;
}>();

function setDefault(value: string | number = '-') {
  return value;
}
</script>

<template>
  <ul :class="$style.trackDetails">
    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Title</p>
      <p ref="title" :class="$style.content">{{ setDefault(track.name) }}</p>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Album</p>

      <div :class="$style.content">
        <NuxtLink
          v-if="track.albumId && track.album"
          :to="`/album/${track.albumId}`"
          :class="$style.link"
        >
          {{ setDefault(track.album) }}
        </NuxtLink>
        <p v-else>-</p>
      </div>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Artists</p>

      <div :class="$style.content">
        <ArtistsList v-if="track.artists.length" :artists="track.artists" />
        <p v-else ref="artistsElse">-</p>
      </div>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Track</p>
      <p :class="$style.content">{{ setDefault(track.trackNumber) }}</p>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Duration</p>
      <time :class="$style.content">
        {{ setDefault(track.duration) }}
      </time>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Year</p>
      <p :class="$style.content">{{ setDefault(track.year) }}</p>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Disc number</p>
      <p :class="$style.content">{{ setDefault(track.discNumber) }}</p>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Genre</p>

      <div :class="$style.content">
        <GenreList v-if="track.genres.length" :genres="track.genres" />
        <p v-else ref="genresElse">-</p>
      </div>
    </li>

    <li :class="$style.detail">
      <p :class="['strong', $style.title]">Size</p>
      <p :class="$style.content">{{ setDefault(track.size) }}</p>
    </li>

    <ul
      v-if="track.information"
      ref="trackInformation"
      :class="$style.trackInformation"
    >
      <li :class="$style.detail">
        <p :class="['strong', $style.title]">Content type</p>
        <p :class="$style.content">
          {{ setDefault(track.information.contentType) }}
        </p>
      </li>

      <li :class="$style.detail">
        <p :class="['strong', $style.title]">Suffix</p>
        <p :class="$style.content">
          {{ setDefault(track.information.suffix) }}
        </p>
      </li>

      <li :class="$style.detail">
        <p :class="['strong', $style.title]">Bitrate</p>
        <p :class="$style.content">
          {{ setDefault(track.information.bitRate) }}
        </p>
      </li>

      <li :class="$style.detail">
        <p :class="['strong', $style.title]">Path</p>
        <p :class="$style.content">{{ setDefault(track.information.path) }}</p>
      </li>
    </ul>
  </ul>
</template>

<style module>
.trackDetails {
  display: table;
  width: 100%;
  border-collapse: collapse;
}

.detail {
  display: table-row;

  &:not(:last-child) {
    border-bottom: 1px solid var(--border-color);
  }

  a {
    color: var(--theme-color);

    &:hover {
      text-decoration: underline;
    }
  }
}

.trackInformation {
  display: table-row-group;
}

.title,
.content {
  display: table-cell;
  padding: var(--space-8) var(--space-16) var(--space-8) 0;
}

.title {
  width: 20%;
}
</style>
