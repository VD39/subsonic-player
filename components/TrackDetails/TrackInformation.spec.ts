import type { VueWrapper } from '@vue/test-utils';
import { mount } from '@vue/test-utils';
import { getFormattedQueueTracksMock } from '@/test/helpers';
import ArtistsList from '@/components/TrackDetails/ArtistsList.vue';
import GenreList from '@/components/TrackDetails/GenreList.vue';
import TrackInformation from './TrackInformation.vue';

const queueTrackMock = getFormattedQueueTracksMock()[0];

function factory(props = {}) {
  return mount(TrackInformation, {
    props: {
      track: queueTrackMock,
      ...props,
    },
  });
}

describe('TrackInformation', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when artists is not an empty array', () => {
    it('shows the ArtistsList component', () => {
      expect(wrapper.findComponent(ArtistsList).exists()).toBe(true);
    });

    it('does not show the else element', () => {
      expect(wrapper.find({ ref: 'artistsElse' }).exists()).toBe(false);
    });
  });

  describe('when artists is an empty array', () => {
    beforeEach(() => {
      wrapper = factory({
        track: {
          ...queueTrackMock,
          artists: [],
        },
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the ArtistsList component', () => {
      expect(wrapper.findComponent(ArtistsList).exists()).toBe(false);
    });

    it('shows the else element', () => {
      expect(wrapper.find({ ref: 'artistsElse' }).exists()).toBe(true);
    });
  });

  describe('when genres is not an empty array', () => {
    it('shows the GenreList component', () => {
      expect(wrapper.findComponent(GenreList).exists()).toBe(true);
    });

    it('does not show the else element', () => {
      expect(wrapper.find({ ref: 'genresElse' }).exists()).toBe(false);
    });
  });

  describe('when genres is an empty array', () => {
    beforeEach(() => {
      wrapper = factory({
        track: {
          ...queueTrackMock,
          genres: [],
        },
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the GenreList component', () => {
      expect(wrapper.findComponent(GenreList).exists()).toBe(false);
    });

    it('shows the else element', () => {
      expect(wrapper.find({ ref: 'genresElse' }).exists()).toBe(true);
    });
  });

  describe('when track information is defined', () => {
    it('shows the track information', () => {
      expect(wrapper.find({ ref: 'trackInformation' }).exists()).toBe(true);
    });
  });

  describe('when track information is not defined', () => {
    beforeEach(() => {
      wrapper = factory({
        track: {
          ...queueTrackMock,
          information: undefined,
        },
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('does not show the track information', () => {
      expect(wrapper.find({ ref: 'trackInformation' }).exists()).toBe(false);
    });
  });

  describe('when a track key is undefined', () => {
    beforeEach(() => {
      wrapper = factory({
        track: {
          ...queueTrackMock,
          title: undefined,
        },
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('sets the default value', () => {
      expect(wrapper.find({ ref: 'title' }).text()).toBe('-');
    });
  });
});
