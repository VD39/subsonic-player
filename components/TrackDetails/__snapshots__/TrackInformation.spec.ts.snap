// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrackInformation > matches the snapshot 1`] = `
"<ul class="trackDetails">
  <li class="detail">
    <p class="strong title">Title</p>
    <p class="content">title</p>
  </li>
  <li class="detail">
    <p class="strong title">Album</p>
    <div class="content"><a class="link">album</a></div>
  </li>
  <li class="detail">
    <p class="strong title">Artists</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="artist-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="artist-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Track</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Duration</p>
    <p class="content">00:19</p>
  </li>
  <li class="detail">
    <p class="strong title">Year</p>
    <p class="content">2024</p>
  </li>
  <li class="detail">
    <p class="strong title">Disc number</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Genre</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="genre-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="genre-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Size</p>
    <p class="content">16</p>
  </li>
  <ul class="trackInformation">
    <li class="detail">
      <p class="strong title">Content type</p>
      <p class="content">contentType</p>
    </li>
    <li class="detail">
      <p class="strong title">Suffix</p>
      <p class="content">suffix</p>
    </li>
    <li class="detail">
      <p class="strong title">Bitrate</p>
      <p class="content">15</p>
    </li>
    <li class="detail">
      <p class="strong title">Path</p>
      <p class="content">path</p>
    </li>
  </ul>
</ul>"
`;

exports[`TrackInformation > when a track key is undefined > matches the snapshot 1`] = `
"<ul class="trackDetails">
  <li class="detail">
    <p class="strong title">Title</p>
    <p class="content">-</p>
  </li>
  <li class="detail">
    <p class="strong title">Album</p>
    <div class="content"><a class="link">album</a></div>
  </li>
  <li class="detail">
    <p class="strong title">Artists</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="artist-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="artist-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Track</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Duration</p>
    <p class="content">00:19</p>
  </li>
  <li class="detail">
    <p class="strong title">Year</p>
    <p class="content">2024</p>
  </li>
  <li class="detail">
    <p class="strong title">Disc number</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Genre</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="genre-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="genre-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Size</p>
    <p class="content">16</p>
  </li>
  <ul class="trackInformation">
    <li class="detail">
      <p class="strong title">Content type</p>
      <p class="content">contentType</p>
    </li>
    <li class="detail">
      <p class="strong title">Suffix</p>
      <p class="content">suffix</p>
    </li>
    <li class="detail">
      <p class="strong title">Bitrate</p>
      <p class="content">15</p>
    </li>
    <li class="detail">
      <p class="strong title">Path</p>
      <p class="content">path</p>
    </li>
  </ul>
</ul>"
`;

exports[`TrackInformation > when artists is an empty array > matches the snapshot 1`] = `
"<ul class="trackDetails">
  <li class="detail">
    <p class="strong title">Title</p>
    <p class="content">title</p>
  </li>
  <li class="detail">
    <p class="strong title">Album</p>
    <div class="content"><a class="link">album</a></div>
  </li>
  <li class="detail">
    <p class="strong title">Artists</p>
    <div class="content">
      <p>-</p>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Track</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Duration</p>
    <p class="content">00:19</p>
  </li>
  <li class="detail">
    <p class="strong title">Year</p>
    <p class="content">2024</p>
  </li>
  <li class="detail">
    <p class="strong title">Disc number</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Genre</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="genre-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="genre-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Size</p>
    <p class="content">16</p>
  </li>
  <ul class="trackInformation">
    <li class="detail">
      <p class="strong title">Content type</p>
      <p class="content">contentType</p>
    </li>
    <li class="detail">
      <p class="strong title">Suffix</p>
      <p class="content">suffix</p>
    </li>
    <li class="detail">
      <p class="strong title">Bitrate</p>
      <p class="content">15</p>
    </li>
    <li class="detail">
      <p class="strong title">Path</p>
      <p class="content">path</p>
    </li>
  </ul>
</ul>"
`;

exports[`TrackInformation > when genres is an empty array > matches the snapshot 1`] = `
"<ul class="trackDetails">
  <li class="detail">
    <p class="strong title">Title</p>
    <p class="content">title</p>
  </li>
  <li class="detail">
    <p class="strong title">Album</p>
    <div class="content"><a class="link">album</a></div>
  </li>
  <li class="detail">
    <p class="strong title">Artists</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="artist-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="artist-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Track</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Duration</p>
    <p class="content">00:19</p>
  </li>
  <li class="detail">
    <p class="strong title">Year</p>
    <p class="content">2024</p>
  </li>
  <li class="detail">
    <p class="strong title">Disc number</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Genre</p>
    <div class="content">
      <p>-</p>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Size</p>
    <p class="content">16</p>
  </li>
  <ul class="trackInformation">
    <li class="detail">
      <p class="strong title">Content type</p>
      <p class="content">contentType</p>
    </li>
    <li class="detail">
      <p class="strong title">Suffix</p>
      <p class="content">suffix</p>
    </li>
    <li class="detail">
      <p class="strong title">Bitrate</p>
      <p class="content">15</p>
    </li>
    <li class="detail">
      <p class="strong title">Path</p>
      <p class="content">path</p>
    </li>
  </ul>
</ul>"
`;

exports[`TrackInformation > when track information is not defined > matches the snapshot 1`] = `
"<ul class="trackDetails">
  <li class="detail">
    <p class="strong title">Title</p>
    <p class="content">title</p>
  </li>
  <li class="detail">
    <p class="strong title">Album</p>
    <div class="content"><a class="link">album</a></div>
  </li>
  <li class="detail">
    <p class="strong title">Artists</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="artist-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="artist-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Track</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Duration</p>
    <p class="content">00:19</p>
  </li>
  <li class="detail">
    <p class="strong title">Year</p>
    <p class="content">2024</p>
  </li>
  <li class="detail">
    <p class="strong title">Disc number</p>
    <p class="content">1</p>
  </li>
  <li class="detail">
    <p class="strong title">Genre</p>
    <div class="content">
      <ul class="meta list">
        <li class="listItem" data-test-id="genre-list-item"><a>name</a></li>
        <li class="listItem" data-test-id="genre-list-item"><a>name1</a></li>
      </ul>
    </div>
  </li>
  <li class="detail">
    <p class="strong title">Size</p>
    <p class="content">16</p>
  </li>
  <!--v-if-->
</ul>"
`;
